Lib/test/test_typing.py

```diff
@@ -3472,6 +3472,12 @@ def meth(self): pass
             vars(NonPR).keys(), vars(D).keys() | acceptable_extra_attrs
         )
 
+    def test_nonruntime_protocols_not_unnecessarily_introspected(self) -> None:
+        class Foo(Protocol):
+            x = 1
+
+        self.assertFalse(hasattr(Foo, "__non_callable_proto_members__"))
```

Comment:
What value does this test add over `test_nonruntime_protocol_interaction_with_evil_classproperty` below? It seems like this one just tests some internal implementation detail that nobody should really care about; the other one tests the behavior we do care about.

Reply:
I think the reason I added this was because the bug that manifested itself to the user was really a symptom of an underlying problem -- and the underlying problem was that we were adding unnecessary attributes to non-runtime protocols and doing unnecessary introspection on them. `test_nonruntime_protocol_interaction_with_evil_classproperty` tests that the symptom is no longer present; this tests that the cause is fixed (which is important in and of itself, since setting unnecessary attributes could potentially lead to other bugs as well).

Having said all that, however, I agree that this doesn't provide much additional value, and is probably more confusing than it is clarifying to people reading through the test suite. So I'll take it out :)